storage/pebble/registers.go

```diff
@@ -53,11 +64,12 @@ func (s *Registers) Get(
 	reg flow.RegisterID,
 	height uint64,
 ) (flow.RegisterValue, error) {
-	latestHeight := s.latestHeight.Load()
-	if height > latestHeight || height < s.firstHeight {
+	latestHeight := s.LatestHeight()
+	firstHeight := s.FirstHeight()
```

Comment:
This two calls reads the `latestHeight` twice, and might potentially return two different values. I would suggest only read `latestHeight` once, and calculate the `firstHeight` by extracting a private function `firstHeight := calculateFirstHeight(latestHeight, s.pruneHeight)`